pkg/destroy/ibmcloud/dedicatedhost.go

```diff
@@ -0,0 +1,225 @@
+package ibmcloud
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	"github.com/pkg/errors"
+)
+
+const (
+	dedicatedHostTypeName      = "dedicated host"
+	dedicatedHostGroupTypeName = "dedicated host group"
+)
+
+// listDedicatedHosts searches for dedicated host that have a name that
+// starts with the cluster's infra ID.
+func (o *ClusterUninstaller) listDedicatedHosts() (cloudResources, error) {
+	o.Logger.Debugf("Listing dedicated hosts")
+	ctx, cancel := o.contextWithTimeout()
+	defer cancel()
+
+	resourceGroupID, err := o.ResourceGroupID()
+	if err != nil {
+		return nil, err
+	}
+
+	options := o.vpcSvc.NewListDedicatedHostsOptions()
+	options.SetResourceGroupID(resourceGroupID)
+	resources, _, err := o.vpcSvc.ListDedicatedHostsWithContext(ctx, options)
+	if err != nil {
+		return nil, err
+	}
+
+	result := []cloudResource{}
```

Comment:
Why collect in an intermediate slice rather than insert each `cloudResource` directly into the `cloudResources` object?

Reply:
I left it as that to keep it consistent with other resources' destroy/list logic

Reply:
Sounds good. I did not check to see whether this was an existing pattern.